### PR TITLE
Add experimental noisyCardinality function

### PR DESCRIPTION
**Pull request by mohandhar**
## Description
Add noisyCardinality function for experimentation.

## Motivation and Context
Unlike cardinality(HyperLogLog), which returns a deterministic approximate count, the noisy_cardinailty(HyperLogLog) function can be used to return a random noisy estimate of cardinality from a HyperLogLog sketch, similar to the random noisy estimates that would be returned from noisy_approx_distinct_sfm.

## Impact
No impact to existing code since we are adding a new function.

## Test Plan
- Manually ran this query within the cli:
```
WITH digests AS (
    SELECT shipdate, APPROX_SET(partkey) AS hll
    FROM tpch.sf1.lineitem
    GROUP BY 1
)

-- This just returns the approximate COUNT(DISTINCT partkey) for each shipdate
SELECT shipdate, NOISY_CARDINALITY(hll, 10)
FROM digests
ORDER BY 1
LIMIT 10;
```

- Added a unit test

## Contributor checklist

- [x] Please make sure your submission complies with our [development](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#development), [formatting](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#formatting), [commit message](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#commit-formatting-and-pull-requests), and [attribution guidelines](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#attribution).
- [x] PR description addresses the issue accurately and concisely.  If the change is non-trivial, a GitHub Issue is referenced.
- [ ] Documented new properties (with its default value), SQL syntax, functions, or other functionality.
- [x] If release notes are required, they follow the [release notes guidelines](https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines).
- [x] Adequate tests were added if applicable.
- [x] CI passed.

## Release Notes
```
== NO RELEASE NOTE ==
```

### CODE REVIEW

**Review by jonhehir — DISMISSED** [review]
Functionality looks good to me. Results will probably be best for larger cardinalities (similar to `noisy_approx_distinct_sfm`), but we can improve on this in the future. Thank you, @mohandhar!

File: presto-main/src/test/java/com/facebook/presto/operator/scalar/TestHyperLogLogFunctions.java [hunk 1]
```diff
@@ -140,6 +140,19 @@ public void testCardinalityManyColumns()
         functionAssertions.assertFunctionWithError(projection, BIGINT, uniqueElements, error);
     }
 
+    @Test
+    public void testNoisyCardinalitySingleColumn()
+    {
+        long uniqueElements = 10000;
+        double error = uniqueElements * 0.05;
+        HyperLogLog hll = HyperLogLog.newInstance(NUMBER_OF_BUCKETS);
```
Comment (mlyublena) [inline presto-main/src/test/java/com/facebook/presto/operator/scalar/TestHyperLogLogFunctions.java:148]: any other test cases that can be added like corner cases etc? For example, for different number of unique elements?

**Comment by steveburnett** [comment]
If we need documentation for this new function, a good place to add it might be [prestodb.io/docs/current/functions/hyperloglog.html ](http://prestodb.io/docs/current/functions/hyperloglog.html).

**Comment by jonhehir** [comment]
> If we need documentation for this new function, a good place to add it might be [prestodb.io/docs/current/functions/hyperloglog.html ](http://prestodb.io/docs/current/functions/hyperloglog.html).

@steveburnett, thanks for the heads up! We're going to punt on documentation for now, but hope to add some soon. We're also planning to add documentation for some related functions shortly (e.g., `noisy_approx_distinct_sfm`), so please stay tuned! (We'll keep you posted for review.) Thanks! 😄

**Review by mlyublena — APPROVED** [review]

**Review by pranjalssh — DISMISSED** [review]
LGTM

**Pull request closed without merging**